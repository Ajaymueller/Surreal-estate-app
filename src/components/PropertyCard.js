import React from 'react';
import '../styles/PropertyCard.css';

const PropertyCard = ({title, type, bathrooms, bedrooms, price, city, email}) => {

        return (
            <div className="property-card">
                <div className="title" data-testid="title-id">
                    {title}
                </div>
                <div className="type" data-testid="type-id">
                    {type} - {city}
                </div>
                <div className="bathrooms" data-testid="bathroom-id">
                    Bathrooms: {bathrooms}
                </div>
                <div className="bedrooms" data-testid="bedroom-id">
                    Bedrooms: {bedrooms}
                </div>
                <div className="price" data-testid="price-id">
                    Price: Â£{price}
                </div>
                <div className="email" data-testid="email-id">
                    <a href={`mailto:${email}`}>Email for more details</a>
                </div>
            </div>
        )
}

export default PropertyCard;
