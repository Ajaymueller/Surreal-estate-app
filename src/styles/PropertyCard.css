
.property-card {
    display: inline-block;
    width: 29.33%;
    padding: 10px;
    border: 5px solid white;
    border-radius: 10px;
    box-sizing: border-box;
    font-size: 20px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    text-align: center;
    background-color: #f1f1f1;
    color:#262626;
    margin: 20px;
  }

.title {
    font-size: larger;
    font-weight: bold;
}


@media screen and (max-width: 600px) {
    .column {
      width: 100%;
      display: block;
      margin-bottom: 20px;
    }
  }