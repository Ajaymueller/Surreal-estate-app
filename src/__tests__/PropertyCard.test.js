import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PropertyCard from '../components/PropertyCard';

const props = {
    title: "randomTitle", 
    type:"Semi-Detached", 
    bathrooms: 3, 
    bedrooms: 4, 
    Price: 200000, 
    Email: "random@email.com"
}; 

describe("PropertyCard", () => {
    it("renders with correct props", () => {
    const { asFragment } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> )
    expect(asFragment).toMatchSnapshot();
    });
    it("has all correct fields", () => {
    const { getByTestId } = render(
    <MemoryRouter><PropertyCard {...props} /></MemoryRouter> )
    expect(getByTestId("title-id")).toHaveClass("title");
    expect(getByTestId("type-id")).toHaveClass("type");
    expect(getByTestId("bathroom-id")).toHaveClass("bathrooms");
    expect(getByTestId("bedroom-id")).toHaveClass("bedrooms");
    expect(getByTestId("price-id")).toHaveClass("price");
    expect(getByTestId("email-id")).toHaveClass("email");
    });
});
