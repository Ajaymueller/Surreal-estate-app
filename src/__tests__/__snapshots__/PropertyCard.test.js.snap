// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders with correct props 1`] = `[Function]`;
